### PR TITLE
Feature/async button

### PR DESCRIPTION
**Pull request by pandafulmanda**
moving from #388 diffs off of alpha to async button off of master

## Waiting (Loading or Saving):
image
screen shot 2015-06-11 at 7 21 53 pm

Error or Timed-out state:
screen shot 2015-06-12 at 4 26 32 pm

### CODE REVIEW

File: src/api.coffee [hunk 1]
```diff
@@ -71,7 +71,7 @@ apiHelper = (Actions, listenAction, successAction, httpMethod, pathMaker) ->
         statusCode = jqXhr.status
         if statusCode is 200
           # HACK For PUT returning nothing (actually, it returns HTML for some reason)
-          successAction(args...)
+          successAction('', args...)
```
Comment (philschatz) [inline src/api.coffee]: Does this additional arg break other success actions?

File: src/components/buttons/async-button.cjsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {isTimedout} = @state
+
+    if isWaiting and not isTimedout
+      _.delay =>
+        @checkForTimeout()
+      , timeout
+
+  checkForTimeout: ->
+    {isWaiting} = @props
+    @setState(isTimedout: true) if isWaiting
```
Comment (philschatz) [inline src/components/buttons/async-button.cjsx]: Wouldn't `isWaiting` always be true since it's a prop?

Comment (pandafulmanda) [inline src/components/buttons/async-button.cjsx]: no, it's a prop that is being updated by the parent render

File: src/api.coffee (hunk 1 above)
Comment (pandafulmanda) [inline src/api.coffee]: i checked, but i can check again.  this should only be happening when we are faking PUTs and PATCHes on our end, and the correct/expected behavior is that ```args...``` is the second argument to follow the pattern set throughout the api helper as in line 68.

File: src/flux/helpers.coffee [hunk 3]
```diff
@@ -110,6 +110,9 @@ CrudConfig = ->
       _.extend(@_changed[id], obj)
       @emitChange()
 
+    _save: (id) ->
+      @_asyncStatus[id] = SAVING
+
```
Comment (philschatz) [inline src/flux/helpers.coffee]: Do we need to emit a change?

Comment (pandafulmanda) [inline src/flux/helpers.coffee]: no, the emitChange is being handled in functions that call this

**Pull request merged**